databricks/koalas/frame.py

```diff
@@ -6810,6 +6810,213 @@ def filter(self, items=None, like=None, regex=None, axis=None):
+        level : int or level name, default None
+            In case of a MultiIndex, only rename labels in the specified level.
+        errors : {'ignore', 'raise}, default 'ignore'
+            If 'raise', raise a `KeyError` when a dict-like `mapper`, `index`, or `columns`
+            contains labels that are not present in the Index being transformed. If 'ignore',
+            existing keys will be renamed and extra keys will be ignored.
+
+        Returns
+        -------
+        DataFrame with the renamed axis labels.
+
+        Raises:
+        -------
+        `KeyError`
+            If any of the labels is not found in the selected axis and "errors='raise'".
+
+        Examples
+        --------
+        >>> kdf1 = ks.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
+        >>> kdf1.rename(columns={"A": "a", "B": "c"}) # doctest: +NORMALIZE_WHITESPACE
+           a  c
+        0  1  4
+        1  2  5
+        2  3  6
+
+        >>> kdf1.rename(index={1: 10, 2: 20}) # doctest: +NORMALIZE_WHITESPACE
+            A  B
+        0   1  4
+        10  2  5
+        20  3  6
+
+        >>> kdf1.rename(str.lower, axis='columns') # doctest: +NORMALIZE_WHITESPACE
+           a  b
+        0  1  4
+        1  2  5
+        2  3  6
+
+        >>> kdf1.rename(lambda x: x*10, axis='index') # doctest: +NORMALIZE_WHITESPACE
+            A  B
+        0   1  4
+        10  2  5
+        20  3  6
+
+        >>> idx = pd.MultiIndex.from_tuples([('X', 'A'), ('X', 'B'),('Y', 'C'), ('Y', 'D')])
+        >>> kdf2 = ks.DataFrame([[1, 2, 3, 4], [5, 6, 7, 8]], columns=idx)
+        >>> kdf2.rename(columns=str.lower, level=0) # doctest: +NORMALIZE_WHITESPACE
+           x     y
+           A  B  C  D
+        0  1  2  3  4
+        1  5  6  7  8
+
+        >>> kdf3 = ks.DataFrame([[1, 2], [3, 4], [5, 6], [7, 8]], index=idx, columns=list('ab'))
+        >>> kdf3.rename(index=str.lower) # doctest: +NORMALIZE_WHITESPACE
+             a  b
+        x a  1  2
+          b  3  4
+        y c  5  6
+          d  7  8
+        """
+
+        if mapper:
+            if axis == 'index' or axis == 0:
+                is_index_mapper = True
+            elif axis == 'columns' or axis == 1:
+                is_index_mapper = False
+            else:
+                raise ValueError("argument axis should be either the axis name "
+                                 "(‘index’, ‘columns’) or number (0, 1)")
+        else:
+            if index:
+                mapper = index
+                is_index_mapper = True
+            elif columns:
+                mapper = columns
+                is_index_mapper = False
+            else:
+                raise ValueError("rename should take either `index` or `columns` mapper.")
+
+        if inplace is None:
+            do_copy = copy
+        else:
+            do_copy = not inplace
+
+        if not do_copy:
+            raise RuntimeError("Koalas dataframe rename method do not support in-place operation.")
```

Comment:
Why do we not support in-place?

Reply:
My understanding is, the underlying spark df is immutable, so in any case the modification should not be called "in-place"

Reply:
Usually we just replace its `internal` with the new one, e.g., https://github.com/databricks/koalas/blob/daa0ccd5673c300e53b1d27863406cf360adbb54/databricks/koalas/frame.py#L2037-L2040